rfcs/rfc-0000-discriminators.md

```diff
@@ -0,0 +1,208 @@
+
+- > Your friend says they changed their name to “vernacular” but actually it’s “𝖛𝖊𝖗𝖓𝖆𝖈𝖚𝖑𝖆𝖗” and you have trouble finding them.
+
+  While this is a valid concern, I would personally put this down as the user's own issue.
+
+# Rationale and alternatives
+
+Discriminators appear to show the least disadvantages out of all of the solutions discussed so far.
+
+| Solution                                                                | Description                                                                                                           | Users have desired username | Selling disincentivized | Privacy | Usability | Difficulty |
+| ----------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------- | :-------------------------: | :---------------------: | :-----: | :-------: | :--------: |
+| **Discriminators**                                                      | Proposed in this RFC                                                                                                  |             ✅              |           ✅            |   ✅    |    ⚠️     |    Low     |
+| **Discriminators**<br/>w/ display names                                 | Show display names with greater priority to username / discriminator combinations.<br/>\* May be proposed in this RFC |             ✅              |           ✅            |   ✅    |    ⚠️     |    Low     |
+| **Discriminators**<br/>w/ display names<br/>w/ restricted character set | Also restrict the characters that you can use in the username itself.<br/>\* May be proposed in this RFC              |             ✅              |           ✅            |   ✅    |    ✅     |  Medium †  |
+| Unique usernames                                                        | Current system on Revolt<br/>(but any unicode username is allowed)                                                    |             ❌              |           ❌            |   ❌    |    ⚠️     |    Low     |
+| Unique usernames<br/>w/ restricted character set                        | Alphanumeric unique global usernames                                                                                  |             ❌              |           ❌            |   ❌    |    ✅     |  Medium †  |
+| Unique usernames<br/>w/ display names                                   | Either unique username solution but with added display names that show instead of the username                        |             ⚠️              |           ❌            |   ⚠️    |    ✅     |    Low     |
+| Remove usernames altogether<br/>(only display names)                    | Resort to using friend codes, invite links, and the like exclusively.                                                 |             ✅              |           ✅            |   ✅    |    ⚠️     |   High ¶   |
```

Comment:
```suggestion
| Solution                                                                | Description                                                                                                           | Users have desired username | Selling disincentivized | Privacy | Usability | Difficulty |
| ----------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------- | :-------------------------: | :---------------------: | :-----: | :-------: | :--------: |
| **Discriminators**                                                      | Proposed in this RFC                                                                                                  |             ✅              |           ✅            |   ✅    |    ⚠️     |    Low     |
| **Discriminators**<br/>w/ display names                                 | Show display names with greater priority to username / discriminator combinations.<br/>\* May be proposed in this RFC |             ✅              |           ✅            |   ✅    |    ⚠️     |    Low     |
| **Discriminators**<br/>w/ display names<br/>w/ restricted character set | Also restrict the characters that you can use in the username itself.<br/>\* May be proposed in this RFC              |             ✅              |           ✅            |   ✅    |    ✅     |  Medium †  |
| Unique usernames                                                        | Current system on Revolt<br/>(but any unicode username is allowed)                                                    |             ❌              |           ❌            |   ❌    |    ⚠️     |    Low     |
| Unique usernames<br/>w/ restricted character set                        | Alphanumeric unique global usernames                                                                                  |             ❌              |           ❌            |   ❌    |    ✅     |  Medium †  |
| Unique usernames<br/>w/ display names                                   | Either unique username solution but with added display names that show instead of the username                        |             ⚠️              |           ❌            |   ⚠️    |    ✅     |    Low     |
| Remove usernames altogether<br/>(only display names)                    | Resort to using friend codes, invite links, and the like exclusively.                                                 |             ✅              |           ✅            |   ✅    |    ⚠️     |   High ¶   |
| Unique cryptographic user identifiers with display names          | Resort to using friend codes, invite links, and the like exclusively.                                                 |              ⚠️              |           ✅            |    ⚠️    |    ⚠️     |   Medium   |
```

Reply:
can you please elaborate? what are cryptographic user ids?

Reply:
in that case, what makes this different from the previous "display names only" solution?

Reply:
@JackDotJS Cryptographic public-private keypairs as user IDs, public part being used to find the user, private part being used as a login.

Reply:
using a cryptographic key for logging in *by default* sounds like a great way to make your platform borderline unusable for most of the world...

keep in mind, most people don't even bother writing down their passwords and just use the same one for every account they use online.

Reply:
+1; encouraging people to adopt good security basics like using a decent password manager and strong, unique passwords are hard enough - using a *cryptographic key* to login seems like a pretty tough sell to say the least

Reply:
Merged for completeness.